test/e2e/clusterctl_upgrade.go

```diff
@@ -328,12 +340,12 @@ func ClusterctlUpgradeSpec(ctx context.Context, inputGetter func() ClusterctlUpg
 		// so we are getting a template using the downloaded version of clusterctl, applying it, and wait for machines to be provisioned.
 
 		workLoadClusterName = fmt.Sprintf("%s-%s", specName, util.RandomString(6))
-		kubernetesVersion := input.E2EConfig.GetVariable(KubernetesVersion)
+		workLoadKubernetesVersion := input.E2EConfig.GetVariable(WorkloadKubernetesVersion)
```

Comment:
This seems to be an additional variable and the one from above is not used?

Reply:
Sorry, it was messed up locally and pushed here as well, removed it